tests/app/main/test_services.py

```diff
@@ -2460,6 +2491,113 @@ def test_shows_service_warning_in_correct_conditions(
+    ):
+        data_api_client.get_framework.return_value = self.framework(slug='digital-outcomes-and-specialists-3')
+        data_api_client.find_draft_services.return_value = {"services": []}
+
+        res = self.client.post(
+            '/suppliers/frameworks/digital-outcomes-and-specialists-3/submissions/digital-outcomes/previous-services',
+            data={'copy': False, 'service_id': '0001'},
+        )
+
+        assert res.status_code == 302
+        assert '/suppliers/frameworks/digital-outcomes-and-specialists-3/submissions/digital-outcomes' in res.location
+        assert copy_service_from_previous_framework.call_args_list == []
+        assert data_api_client.create_new_draft_service.call_args_list == [
+            mock.call('digital-outcomes-and-specialists-3', 'digital-outcomes', 1234, {}, 'email@email.com')
+        ]
+
+    def test_400s_if_lot_does_not_have_one_service_limit(self, data_api_client):
+        data_api_client.get_framework.return_value = self.framework(slug='digital-outcomes-and-specialists-3')
+
+        res = self.client.post(
+            '/suppliers/frameworks/digital-outcomes-and-specialists-3/'
+            'submissions/user-research-studios/previous-services',
+            data={'copy': False, 'service_id': '0001'},
+        )
+
+        assert res.status_code == 400
+
+    def test_400s_if_draft_already_exists(self, data_api_client):
+        data_api_client.get_framework.return_value = self.framework(slug='digital-outcomes-and-specialists-3')
+        data_api_client.find_draft_services.return_value = {
+            "services": [
+                {'status': 'not-submitted', 'lotSlug': 'digital-specialists'},
+            ]
+        }
+
+        res = self.client.post(
+            '/suppliers/frameworks/digital-outcomes-and-specialists-3/'
+            'submissions/digital-specialists/previous-services',
+            data={'copy': False, 'service_id': '0001'},
+        )
+
+        assert res.status_code == 400
+
+    def test_validates_that_an_answer_is_required(self, data_api_client):
+        data_api_client.get_framework.return_value = self.framework(slug='digital-outcomes-and-specialists-3')
+        data_api_client.find_draft_services.return_value = {"services": []}
+        data_api_client.find_services.return_value = {"services": [{'copiedToFollowingFramework': False}]}
+
+        res = self.client.post(
+            '/suppliers/frameworks/digital-outcomes-and-specialists-3/'
+            'submissions/digital-specialists/previous-services',
+            data={'service_id': '0001'},
+        )
+        doc = html.fromstring(res.get_data(as_text=True))
+
+        assert res.status_code == 200
+        assert "Do you want to reuse your previous digital specialists service?" == doc.xpath(
+            "//a[@class='validation-masthead-link']/text()"
+        )[0].strip()
+        assert "You must answer this question." == doc.xpath(
```

Comment:
If "you must answer this question" is a constant somewhere, then it'd be better to reference that constant.

Reply:
It's not really, it's defined in the form as the validation message for that particular field. Not sure how easy it would be to import.

Reply:
You could define the message as a class variable for the form and then use that here easily enough with something like `OneServiceLimitCopyServiceForm.ANSWER_REQUIRED_MESSAGE`.